### DISCUSSION TITLE
The subscriber would not send DATA (r) after receiving ACKNACK  while in EDP

### DISCUSSION THREAD

**Discussion by xukunfeng0496** [eProsima/Fast-DDS#2953] [Q&A]
When we use ros2 based on fastdds v2.2.1, we have subscriber sub1 and publisher pub1。We found occasional probabilistic subscriber sub1 unable to receive messages, and unable to recover unless restarted.
When this phenomenon occurred, we made the following attempts:
1. Create a new publisher pub2 that publishes the same topic, and find that sub1 can receive messages
2. Create a new subscriber sub2 and find that sub2 can receive messages from pub1.

After we captured the package, 
Publisher: rtps.appId:0x08097693,udp.port:7418/7419
Subscriber: rtps.appId:0x770974c1,udp.port:7428/7429
TopicName: rt/encoding

We found that,during the EDP discovery phase, the publisher continues to send ACKNACK to the subscriber, and the subscriber has not replied DATA(r) to tell the publisher.

<img width="1567" alt="image" src="https://user-images.githubusercontent.com/62683570/190374879-979f4aa7-6f8e-456c-805f-7079cbaba95e.png">
 I'm not so familiar with FastRTPS lower layer, it will be so nice of you to help me ! Thanks!

[bad.pcap.zip](https://github.com/eProsima/Fast-DDS/files/9573988/bad.pcap.zip)

**Comment by EduPonz** [comment]
Hi @xukunfeng0496 ,

I'm afraid all v2.2.x versions are EOL and therefore do not receives support. I'm guessing that you are building ROS 2 from sources, since Fast DDS v2.2.1 is not shipped with any ROS 2 distribution. In that case, could you check with a newer version of Fast DDS?

**Comment by xukunfeng0496** [comment]
sorry  , the version is v2.1.1

**Comment by EduPonz** [comment]
Then I assume you're using ROS 2 Foxy, for which the latest version of Fast DDS is v2.1.2. Could you try upgrading Foxy and check again?

**Comment by xukunfeng0496** [comment]
We are trying to upgrade to ROS2 humble version to reproduce. Can you see any clues from the above wireshark packet capture?
